### DISCUSSION TITLE
Finding SHA or ref of branch for [push] event after a pull-request

### DISCUSSION THREAD

**Discussion by ghutchis** [community/community#26459] [Actions]
<p>I’m trying to make an action workflow that operates on the <code>git diff</code> from the branch point.</p>
<p>The current SHA is easy:</p>
<pre><code>github.sha
</code></pre>
<p>When I have a pull-request event, I can also use <code>github.head_ref</code> to get the source branch of the pull request.</p>
<p>But when I push to that branch (e.g. fixing something), then <code>github.head_ref</code> is empty.</p>
<p>Is there a way to continue to access <code>github.head_ref</code> on push events already part of a pull request? I’d accept some git methods, something like:</p>
<pre><code>git merge-base --fork-point origin/master
</code></pre>
<p>But I see that there’s no .git/refs/heads/master when I check out…</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/ghutchis">@ghutchis</a> ,</p>
<p>‘github.head_ref’ is not supported for push event, since it’s not in the push event github <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">context</a>, you can use below command to check all of the context values:</p>
<pre><code>- name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
</code></pre>
<p>Hope it helps!</p>

**Comment by ghutchis** [comment, marked as answer]
<p>I think this is a fairly common use case, judging from some of the searching I did. I think push events made to a pull request should be able to fetch their branch point.</p>
<p>This doesn’t work robustly, but is close enough for now:</p>
<pre><code># I didn't use ${{ github.repository }} because that points to the fork
git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/openchemistry/avogadrolibs

git fetch origin master:master
export MASTER_SHA=`cat .git/refs/heads/master`
git diff ${MASTER_SHA} --name-only
# whatever else you want to do
</code></pre>

**Reply by kirkath** [reply]
dont think we need to pull down whole master for this.

```
git ls-remote https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/openchemistry/avogadrolibs --tags master | sed "s/\tHEAD//" | cut -d $'\t' -f 1
```

will do the trick
